components/eventing-controller/pkg/backend/jetstreamv2/connection.go

```diff
@@ -0,0 +1,45 @@
+package jetstreamv2
+
+import (
+	backendnats "github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/nats"
+	pkgerrors "github.com/kyma-project/kyma/components/eventing-controller/pkg/errors"
+	"github.com/nats-io/nats.go"
+)
+
+type ConnectionBuilderInterface interface {
+	Build() (ConnectionInterface, error)
+}
+
+// ConnectionBuilder helps in establishing a connection to NATS.
+type ConnectionBuilder struct {
+	config backendnats.Config
+}
+
+func NewConnectionBuilder(config backendnats.Config) ConnectionBuilder {
+	return ConnectionBuilder{config: config}
+}
+
+// Build connects to NATS and returns the connection. If an error occurs, ErrConnect is returned.
+func (b ConnectionBuilder) Build() (ConnectionInterface, error) {
+	config := b.config
+	jsOptions := []nats.Option{
+		nats.RetryOnFailedConnect(true),
+		nats.MaxReconnects(config.MaxReconnects),
+		nats.ReconnectWait(config.ReconnectWait),
+		nats.Name("Kyma Controller"),
+	}
+	conn, err := nats.Connect(config.URL, jsOptions...)
+	if err != nil || !conn.IsConnected() {
+		return nil, pkgerrors.MakeError(ErrConnect, err)
+	}
+
+	return conn, nil
+}
+
+// ConnectionInterface is a contract for a NATS connection object.
+type ConnectionInterface interface {
```

Comment:
I am thinking about the name. I know there is this `-er` convention for Interfaces, but I am also not always super convinced about that. 

I like the approach in [this](https://medium.com/@dotronglong/interface-naming-convention-in-golang-f53d9f471593) short article; especially what is written at point # 3. Let me know what you think.